.github/workflows/workflow.yaml

```diff
@@ -0,0 +1,153 @@
+name: Workflow for rke-tools
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - master
+
+env:
+  IMAGE: rancher/rke-tools
+
+jobs:
+  ci:
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    container: 
+      image: rancher/dapper:v0.6.0
+    steps:
+      - name: Force Install GIT latest
```

Comment:
Just to check my understanding, is this step needed because we are using the dapper image for this job?

Reply:
No, its needed because we are running git commands as part of the `version` script. Without this step the `.git` folder does not get downloaded on the runner and all `git` commands start failing with not a git repository error.
ref: https://github.com/actions/checkout/issues/363

Reply:
I saw the `ci` jon worked [here](https://github.com/mitulshah-suse/rke-tools/actions/runs/9116499897), but I am still wondering how this step helps because dapper downloads git in its [Docker.dapper](https://github.com/rancher/rke-tools/blob/master/Dockerfile.dapper#L6) file instead of using the global git.

Reply:
If we dont have this step, it does not download the `.git` folder and any git commands that we run end up failing with `not a git repository` error. The underlying problem is that for some reason, the checkout needs git binary in the PATH and at a version higher than 2.18, for it to work correctly and download the `.git` folder.

Reply:
OK, I think I understand the necessity now. Dapper copies or mounts the source code from the runner to its 
 own container to perform operations, if the source code in the runner does not have the .git folder, Dapper will hit the not-a-git-repo error even if it downloads git by itself, therefore, we need to install git before checking out the code in the runner. 

@mitulshah-suse, one more question: why don't you need to run the command `git config --global --add safe.directory $(realpath .)` after installing git per the recommendation in the [GH issue](https://github.com/actions/checkout/issues/363#issuecomment-1163942488)?

Reply:
I did not see any permission issue when running it, so did not need to add that step.